core/src/lib/base64.cc

```diff
@@ -219,3 +219,22 @@ int Base64ToBin(char* dest, int dest_size, char* src, int srclen)
 
   return (bufout - (uint8_t*)dest);
 }
+
+/**
+ * Calculate the correct unpadded base64 length for a given length
+ * of an unencoded string
+ *
+ *  Returns: Length of unpadded base64
+ */
+int Base64LengthUnpadded(const int source_length)
+{
+  if (source_length == 0) { return 0; }
+
+  int quotient = source_length / 3;
+  const int remainder = source_length % 3;
+  if (remainder > 0) { ++quotient; }
+  int unpadded_length = 4 * quotient;
+  if (remainder > 0) { unpadded_length -= 3 - remainder; }
```

Comment:
While the operator precedence is  well defined, `unpadded_length -= (3 - remainder)` would make it unambiguous to the reader at first sight (and not only the compiler)